### ISSUE TITLE
Add pyflakes/pylint/pychecker to CI for python files

### ISSUE THREAD

**Issue by cagery** [ROCmSoftwarePlatform/AMDMIGraphX#897]
For avoiding coding malpractices 

cc: @pfultz2

**Comment by pfultz2** [comment]
Here is the output from pyflakes, ran with `pyflakes examples/ tools/ src/ test/ doc/`:

```
examples/python_bert_squad_example/bert-squad-migraphx.py:3:1 'time' imported but unused
examples/python_bert_squad_example/bert-squad-migraphx.py:5:1 'os.path' imported but unused
examples/python_bert_squad_example/bert-squad-migraphx.py:6:1 'sys' imported but unused
examples/python_bert_squad_example/bert-squad-migraphx.py:9:1 'from run_onnx_squad import *' used; unable to detect undefined names
examples/python_bert_squad_example/bert-squad-migraphx.py:23:17 'read_squad_examples' may be undefined, or defined from star imports: run_onnx_squad
examples/python_bert_squad_example/bert-squad-migraphx.py:36:50 'convert_examples_to_features' may be undefined, or defined from star imports: run_onnx_squad
examples/python_bert_squad_example/bert-squad-migraphx.py:74:13 'RawResult' may be undefined, or defined from star imports: run_onnx_squad
examples/python_bert_squad_example/bert-squad-migraphx.py:82:1 'write_predictions' may be undefined, or defined from star imports: run_onnx_squad
examples/python_bert_squad_example/run_onnx_squad.py:32:1 'logging' imported but unused
examples/python_bert_squad_example/run_onnx_squad.py:148:9 local variable 'tok_start_position' is assigned to but never used
examples/python_bert_squad_example/run_onnx_squad.py:149:9 local variable 'tok_end_position' is assigned to but never used
examples/python_bert_squad_example/run_onnx_squad.py:570:40 undefined name 'vocab_file'
tools/api.py:1:1 'os' imported but unused
tools/te.py:1:1 'os' imported but unused
src/py/backend/backend.py:86:28 undefined name 'options'
src/py/backend/backend.py:87:29 undefined name 'options'
src/py/backend/__init__.py:6:1 '.backend.is_compatible' imported but unused
src/py/backend/__init__.py:6:1 '.backend.prepare' imported but unused
src/py/backend/__init__.py:6:1 '.backend.run' imported but unused
src/py/backend/__init__.py:6:1 '.backend.supports_device' imported but unused
test/py/onnx_backend_test.py:7:1 'platform' imported but unused
test/py/test_op.py:1:1 'array' imported but unused
test/py/test_op.py:1:1 'sys' imported but unused
test/tf/gen_tf_pb.py:4:1 'numpy as np' imported but unused
test/tf/gen_tf_pb.py:6:1 'tensorflow.core.framework.attr_value_pb2' imported but unused
test/onnx/gen_onnx.py:7:1 'onnx.numpy_helper' imported but unused
test/onnx/gen_onnx.py:8:1 'onnx.AttributeProto' imported but unused
test/onnx/gen_onnx.py:8:1 'onnx.GraphProto' imported but unused
test/onnx/gen_onnx.py:486:5 local variable 'shape' is assigned to but never used
test/onnx/gen_onnx.py:537:5 local variable 'shape_val' is assigned to but never used
test/onnx/gen_onnx.py:1613:5 local variable 'empty_out' is assigned to but never used
test/onnx/gen_onnx.py:2297:5 local variable 'y' is assigned to but never used
test/onnx/gen_onnx.py:3193:5 local variable 'trans_x' is assigned to but never used
test/onnx/gen_onnx.py:3437:5 local variable 'y' is assigned to but never used
test/onnx/gen_onnx.py:3744:5 local variable 'y' is assigned to but never used
test/onnx/gen_onnx.py:3859:5 local variable 'y' is assigned to but never used
```

We need to fix these issues before we can enable pyflakes. 

I'll look into fixing the issues under tools/. @cagery Are you able to fix the issues in the example? @scxiao Can you look into fixing the onnx backend issues? @kahmed10 Can you look into the tf and onnx python generators?

**Comment by cagery** [comment]
working on `/examples/` in `nb_clean` branch #898

**Comment by scxiao** [comment]
fixed onnx_backend_test related warnings and created PR #901

**Comment by causten** [comment]
Team determined that pyFlakes will suffice as a solution.  pylint requires all libraries to be available which is beyond the scope of our CI goals at this time.  pyflakes was merged with #902
